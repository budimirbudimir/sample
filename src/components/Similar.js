// @flow

import React from 'react'

import type { SimilarArtist } from '../models'

type SimilarTypes = {
	// similar is array of objects containing similar artist data
	similar: SimilarArtist[],
	// fetchArtist is function to get artist data by artist name
	fetchArtist: string => void,
}

const Similar = ({ similar, fetchArtist }: SimilarTypes) => (
	<p>
		Similar acts:
		{similar.map((act, index) => (
			<span
				key={index}
				className="App-similar_act"
				href={act.url}
				onClick={() => fetchArtist(act.name)}
			>
				{act.name}
			</span>
		))}
	</p>
)

export default Similar
