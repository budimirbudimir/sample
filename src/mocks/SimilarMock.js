export default [
	{ name: 'Lou Reed', url: 'https://www.last.fm/music/Lou+Reed' },
	{ name: 'Iggy Pop', url: 'https://www.last.fm/music/Iggy+Pop' },
	{ name: 'Tin Machine', url: 'https://www.last.fm/music/Tin+Machine' },
	{ name: 'T. Rex', url: 'https://www.last.fm/music/T.+Rex' },
	{ name: 'Roxy Music', url: 'https://www.last.fm/music/Roxy+Music' },
]
