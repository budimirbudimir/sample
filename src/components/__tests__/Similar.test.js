import React from 'react'
import Similar from '../Similar'
import { shallow } from 'enzyme'

import SimilarMock from '../../mocks/SimilarMock'

describe('Similar', () => {
	it('matches snapshot', () => {
		const wrapper = shallow(
			<Similar similar={SimilarMock} fetchArtist={() => null} />,
		)
		expect(wrapper).toMatchSnapshot()
	})
})
