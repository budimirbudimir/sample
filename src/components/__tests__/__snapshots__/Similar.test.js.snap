// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Similar matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Similar
    fetchArtist={[Function]}
    similar={
        Array [
            Object {
              "name": "Lou Reed",
              "url": "https://www.last.fm/music/Lou+Reed",
            },
            Object {
              "name": "Iggy Pop",
              "url": "https://www.last.fm/music/Iggy+Pop",
            },
            Object {
              "name": "Tin Machine",
              "url": "https://www.last.fm/music/Tin+Machine",
            },
            Object {
              "name": "T. Rex",
              "url": "https://www.last.fm/music/T.+Rex",
            },
            Object {
              "name": "Roxy Music",
              "url": "https://www.last.fm/music/Roxy+Music",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="Similar-title"
>
          Similar acts:
</span>,
        Array [
          <span
            className="App-similar_act"
            href="https://www.last.fm/music/Lou+Reed"
            onClick={[Function]}
>
            Lou Reed
</span>,
          <span
            className="App-similar_act"
            href="https://www.last.fm/music/Iggy+Pop"
            onClick={[Function]}
>
            Iggy Pop
</span>,
          <span
            className="App-similar_act"
            href="https://www.last.fm/music/Tin+Machine"
            onClick={[Function]}
>
            Tin Machine
</span>,
          <span
            className="App-similar_act"
            href="https://www.last.fm/music/T.+Rex"
            onClick={[Function]}
>
            T. Rex
</span>,
          <span
            className="App-similar_act"
            href="https://www.last.fm/music/Roxy+Music"
            onClick={[Function]}
>
            Roxy Music
</span>,
        ],
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Similar acts:",
          "className": "Similar-title",
        },
        "ref": null,
        "rendered": "Similar acts:",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": "Lou Reed",
          "className": "App-similar_act",
          "href": "https://www.last.fm/music/Lou+Reed",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Lou Reed",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": "Iggy Pop",
          "className": "App-similar_act",
          "href": "https://www.last.fm/music/Iggy+Pop",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Iggy Pop",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": "Tin Machine",
          "className": "App-similar_act",
          "href": "https://www.last.fm/music/Tin+Machine",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Tin Machine",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "host",
        "props": Object {
          "children": "T. Rex",
          "className": "App-similar_act",
          "href": "https://www.last.fm/music/T.+Rex",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "T. Rex",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": "4",
        "nodeType": "host",
        "props": Object {
          "children": "Roxy Music",
          "className": "App-similar_act",
          "href": "https://www.last.fm/music/Roxy+Music",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Roxy Music",
        "type": "span",
      },
    ],
    "type": "p",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="Similar-title"
>
            Similar acts:
</span>,
          Array [
            <span
              className="App-similar_act"
              href="https://www.last.fm/music/Lou+Reed"
              onClick={[Function]}
>
              Lou Reed
</span>,
            <span
              className="App-similar_act"
              href="https://www.last.fm/music/Iggy+Pop"
              onClick={[Function]}
>
              Iggy Pop
</span>,
            <span
              className="App-similar_act"
              href="https://www.last.fm/music/Tin+Machine"
              onClick={[Function]}
>
              Tin Machine
</span>,
            <span
              className="App-similar_act"
              href="https://www.last.fm/music/T.+Rex"
              onClick={[Function]}
>
              T. Rex
</span>,
            <span
              className="App-similar_act"
              href="https://www.last.fm/music/Roxy+Music"
              onClick={[Function]}
>
              Roxy Music
</span>,
          ],
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Similar acts:",
            "className": "Similar-title",
          },
          "ref": null,
          "rendered": "Similar acts:",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": "Lou Reed",
            "className": "App-similar_act",
            "href": "https://www.last.fm/music/Lou+Reed",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Lou Reed",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": "Iggy Pop",
            "className": "App-similar_act",
            "href": "https://www.last.fm/music/Iggy+Pop",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Iggy Pop",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": "Tin Machine",
            "className": "App-similar_act",
            "href": "https://www.last.fm/music/Tin+Machine",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Tin Machine",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "host",
          "props": Object {
            "children": "T. Rex",
            "className": "App-similar_act",
            "href": "https://www.last.fm/music/T.+Rex",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "T. Rex",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "host",
          "props": Object {
            "children": "Roxy Music",
            "className": "App-similar_act",
            "href": "https://www.last.fm/music/Roxy+Music",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Roxy Music",
          "type": "span",
        },
      ],
      "type": "p",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
